### DISCUSSION TITLE
Weird requirement

### DISCUSSION THREAD

**Discussion by vbav** [react-hook-form/react-hook-form#3473] [Q&A] [answered]
I have a large form where I have about 15 fields. All 15 fields are in one form and I have one submit button. Validation works fine. Today, we got a new requirement that we need to validate the first 5 fields on click of another button which is also inside this form. 
How do we trigger partial validation on these 5 fields. Is there a way I can validate each of these 5 fields separately rather than the submit button?

**Comment by bluebill1049** [comment, marked as answer]
`trigger` api

**Reply by vbav** [reply]
Thanks. I think this is what I was looking for.

**Reply by vbav** [reply]
@bluebill1049 - I am not sure if this is a bug or expected behavior. I used the trigger on 5 fields out of which 3 fields are hidden. But still it is trying to do the validation on the 3 hidden fields also. The workaround I did was to push the 3 hidden fields into the array only when required. Please check.

**Reply by bluebill1049** [reply]
can you share a codesandbox?

**Reply by vbav** [reply]
@bluebill1049 - Sorry for the delay. Here is the codebox:
https://codesandbox.io/s/react-hook-form-watch-v6-forked-82pmq 

If I am doing anything wrong, please let me know.

**Reply by bluebill1049** [reply]
what's the steps to reproduce? also watch is been used 3 times? i think you should just use 1 of them.

**Reply by vbav** [reply]
@bluebill1049 - I have used the codebox solution for the watch api from your website. That is why you see watch statement thrice. I have removed it now. Just try click on "Validate All" button. Should it not return true if you fill in first name and click on Validate ALL button since the age field is hidden?

Click on Submit button however, validates the form without having to fill in the age field if it is hidden, successfully which is the correct behavior.

**Reply by bluebill1049** [reply]
Your age input is never getting registered at hook form right? because it's not even rendered.

**Reply by vbav** [reply]
@bluebill1049 - Correct, since the age control is not even rendered, shouldn't the trigger api ignore it and returned true?

**Reply by bluebill1049** [reply]
you should exclude that in the trigger if there is no such input tho.

**Reply by vbav** [reply]
Oh ok.. So, it is expected behavior from trigger API?
